core/src/main/resources/jenkins/views/JenkinsHeader/headerContent.jelly

```diff
@@ -1,78 +1,8 @@
-                <!-- this div is used to calculate the width of the text box -->
-                <div id="search-box-sizer"/>
-                <div id="searchform">
-                    <input name="q" placeholder="${searchPlaceholder}" id="search-box" class="main-search__input" value="${request.getParameter('q')}" role="searchbox" />
-
-                    <span class="main-search__icon-leading">
-                        <l:icon src="symbol-search"/>
-                    </span>
-                    <a href="${searchHelpUrl}" class="main-search__icon-trailing">
-                        <l:icon src="symbol-help-circle"/>
-                    </a>
-
-                    <div id="search-box-completion" data-search-url="${searchURL}" />
-                    <st:adjunct includes="jenkins.views.JenkinsHeader.search-box" />
-                </div>
-            </form>
-        </div>
-
-        <div class="login page-header__hyperlinks">
-            <div id="visible-am-insertion" class="page-header__am-wrapper" />
-            <div id="visible-sec-am-insertion" class="page-header__am-wrapper" />
-
-            <!-- user icon and login/logout links; only show if authentication is enabled and we're not handling a servlet error -->
-            <j:if test="${app.useSecurity}">
-                <j:choose>
-                    <j:when test="${!h.isAnonymous()}">
-                        <j:invokeStatic var="user" className="hudson.model.User" method="current" />
-                        <j:choose>
-                            <j:when test="${user.fullName == null || user.fullName.trim().isEmpty()}">
-                                <j:set var="userName" value="${user.id}"/>
-                            </j:when>
-                            <j:otherwise>
-                                <j:set var="userName" value="${user.fullName}"/>
-                            </j:otherwise>
-                        </j:choose>
-                        <a href="${rootURL}/${user.url}" class="model-link">
-                            <l:icon src="symbol-person-circle" class="icon-md"/>
-                            <span class="hidden-xs hidden-sm">${userName}</span>
-                        </a>
-                        <j:if test="${app.securityRealm.canLogOut()}">
-                            <a href="${rootURL}/logout">
-                                <l:icon src="symbol-log-out" class="icon-md" />
-                                <span class="hidden-xs hidden-sm">${logout}</span>
-                            </a>
-                        </j:if>
-                    </j:when>
-                    <j:otherwise>
-                        <st:include it="${app.securityRealm}" page="loginLink.jelly" />
-                    </j:otherwise>
-                </j:choose>
-            </j:if>
-        </div>
+      <h:logo/>
+      <h:searchbox/>
+      <h:login/>
```

Comment:
This is now public API. Are we confident a redesign doesn't just randomly change this anytime soon?

Reply:
Any substantial change to the header would require a change of the header version in the [PartialHeader](https://github.com/jenkinsci/jenkins/blob/b89863cf1f1ca7a27626774bd713f9a1d6fde2f6/core/src/main/java/jenkins/views/PartialHeader.java#L28C30-L28C57) so that plugins need to react. I might need to stick to the PartialHeader in the PR of my plugin (which is so far the only implementation of a header) https://github.com/jenkinsci/customizable-header-plugin/blob/295683015391e800be41c568b42ac80f6fd7c61b/src/main/java/io/jenkins/plugins/customizable_header/headers/LogoHeader.java#L9